src/libraries/System.Private.CoreLib/src/System/Resources/ResourceReader.cs

```diff
@@ -915,42 +918,56 @@ private void _ReadResources()
         // This allows us to delay-initialize the Type[].  This might be a
         // good startup time savings, since we might have to load assemblies
         // and initialize Reflection.
+        [UnconditionalSuppressMessage("ReflectionAnalysis", "IL2026:RequiresUnreferencedCode",
+            Justification = "UseReflectionToGetType will get trimmed out when AllowCustomResourceTypes is set to false. " +
+            "When set to true, we will already throw a warning for this feature switch, so we suppress this one in order for" +
+            "the user to only get one error.")]
         private Type FindType(int typeIndex)
         {
+            if (!AllowCustomResourceTypes)
```

Comment:
I think this method is called by `V1`. I think that's fine. But I just wanted to check, as I'm not an expert on resources.

Reply:
Great catch, I thought it would work as well from V1 but I checked again and that is not the case, so I reverted @jkotas's refactoring suggestion to be compatible with v1 resources.

Reply:
How does this revert make the V1 resources work?

As far as I can tell, the V1 resources are always going to hit the reflection path and throw, both before and after the last revert.

Reply:
I think it is fine for the V1 resources to throw. They would only show up in .dlls built with a very ancient tooling. In a separate PR, it may be a good idea to delete the V1 resources path completely.

Reply:
You are right @jkotas, I swear I thought I saw _typeTable[typeIndex] was assigned somewhere else besides the reflection code, but that doesn't seem to be the case, so it should be safe to assume that the table will be empty when entering this method, which for V1 that is the very first thing that happens. That means that as you suggested, V1 resources will basically be unsafe no matter what when trimming, so I'll revert the refactoring back to what you pointed out in order to trim additional code.